File: src/string-calculator.js
"use strict";

class Calculator {
    constructor() {
        this._defaultValue = 0;
    }

    add(numbers) {
        if (this._isEmpty(numbers)) {
            return this._defaultValue;
        }

        if (this._isSingleNumber(numbers)) {
            return this._parseSingleNumber(numbers);
        }

        return this._parseTwoNumbers(numbers);
    }

    _parseTwoNumbers(numbers) {
        return numbers.split(',').reduce((a, b) => {
            return this._parseSingleNumber(a) + this._parseSingleNumber(b)
        });
    }

    _isSingleNumber(numbers) {
        return numbers.indexOf(',') === -1;
    }

    _parseSingleNumber(numbers) {
        return parseInt(numbers, 10);
    }

    _isEmpty(numbers) {
        return !numbers;
    }
}

module.exports = Calculator;

File: homework/string_calculator.py
import re


class Calculator(object):
    def __init__(self, p=None):
        self.separators = ',|\n'
        if p:
            self.separators += '|' + p

    def add(self, numbers):
        res = 0

        if len(numbers) > 4 and numbers[3] == '\n' and numbers[0] == '/' and numbers[1] == '/':
            self.separators += '|' + numbers[2]
            numbers = numbers[3:]

        split_numbers = re.split(self.separators, numbers)

        for i in split_numbers:
            res += self.__parse_int(i)
        return res

    @classmethod
    def __parse_int(cls, numbers):
        if numbers == "":
            return 0
        return int(numbers)
